### PR TITLE
is possible to validate mail

### PR DESCRIPTION
**Pull request by marcinbutora**

### CODE REVIEW

File: training-service/src/app/pages/mail-form/mail-form.component.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+
+  mailForm = new FormGroup({
+    emailAddress: new FormControl("", [
+      Validators.required,
+      Validators.pattern("^[a-z0-9._%+-]+@[a-z0-9.-]+\\.[a-z]{2,4}$")
+    ])
+  });
+
+  ngOnInit(): void {
```
Comment (holistic-fox) [inline training-service/src/app/pages/mail-form/mail-form.component.ts]: ngOnInit is empty please remove.

File: training-service/src/app/pages/mail-form/mail-form.component.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+  templateUrl: './mail-form.component.html',
+  styleUrls: ['./mail-form.component.scss']
+})
+export class MailFormComponent implements OnInit {
+
+
+  mailForm = new FormGroup({
+    emailAddress: new FormControl("", [
+      Validators.required,
+      Validators.pattern("^[a-z0-9._%+-]+@[a-z0-9.-]+\\.[a-z]{2,4}$")
```
Comment (holistic-fox) [inline training-service/src/app/pages/mail-form/mail-form.component.ts]: Please export ^[a-z0-9._%+-]+@[a-z0-9.-]+\\.[a-z]{2,4}$ to a const

File: training-service/src/app/pages/mail-form/mail-form.component.html [hunk 3]
```diff
@@ -0,0 +1,21 @@
+<h2>E-mail Form Validation</h2>
+<form [formGroup]="mailForm">
+  <div class="form-group">
+    <input
+      type="email"
+      class="form-control"
+      placeholder="Enter e-mail"
+      formControlName="emailAddress"
+      [ngClass]="!mailForm.valid ? 'not-valid-input' : 'valid-input'"
```
Comment (holistic-fox) [inline training-service/src/app/pages/mail-form/mail-form.component.html]: Are you sure you do not want to check touched prop also ?

**Review by holistic-fox — CHANGES_REQUESTED** [review]

File: training-service/src/app/pages/mail-form/mail-form.component.ts (hunk 1 above)
Comment (marcinbutora) [reply training-service/src/app/pages/mail-form/mail-form.component.ts]: done.

File: training-service/src/app/pages/mail-form/mail-form.component.ts (hunk 2 above)
Comment (marcinbutora) [reply training-service/src/app/pages/mail-form/mail-form.component.ts]: done.

File: training-service/src/app/pages/mail-form/mail-form.component.html (hunk 3 above)
Comment (marcinbutora) [reply training-service/src/app/pages/mail-form/mail-form.component.html]: done.

**Review by holistic-fox — APPROVED** [review]

**Pull request merged**